### PR TITLE
Honor Host header when connecting to IPs

### LINKED ISSUE #36: (Option to) show which IP is being connected to
My applications often have numerous IPs (CDNs, etc), so showing which IP is being connected to is helpful.

(Along the same lines, an option to connect to a particular IP could also be helpful; along the lines of the -x or --resolv options from curl).

### PR DESCRIPTION
**Pull request by moorereason**
Fixes #36

### CODE REVIEW

**Comment by davecheney** [comment]
I wonder if this is a bridge too far, the feature is quite esoteric, and its usage is even more so. I'm leaning towards saying No to this one.

**Comment by moorereason** [comment]
Let's try this again after getting some sleep.  :wink:

**Comment by theckman** [comment]
@davecheney at work I'm often doing testing against singular IPs, while passing in a `Host` header:

```
curl -H 'Host: blah.work.tld' http://<TEST_IP>/
```

I'd replace my usage with `httpstat` if it supported it.

File: .travis.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ script:
   - ./httpstat -X DELETE http://httpbin.org/delete
   - ./httpstat https://www.google.com/
   - ./httpstat https://www.google.com:443/
+  - ./httpstat -H "Host: www.google.com" https://$(dig +short www.google.com):443/
```
Comment (theckman) [inline .travis.yml]: Can we make this a bit more resistant to failure? If Google ever adds multiple A records this test will fail. How do you feel about `dig +short www.google.com | head -n 1`?

Comment (theckman) [inline .travis.yml]: Oh, nevermind...

```
$ {:"./httpstat -H \"Host"=>"www.google.com\" https://$(dig +short www.google.com):443/"}
/home/travis/build.sh: line 45: dig: command not found
/home/travis/build.sh: line 45: dig: command not found
/home/travis/build.sh: line 45: www.google.com" https://:443/}: No such file or directory
```

😂

**Comment by theckman** [comment]
@moorereason maybe we can add this to the top of the `.travis.yml` file:

```
addons:
  apt:
    packages:
    - dnsutils
```

You should then have `dig`.

**Comment by moorereason** [comment]
@theckman, thanks for the addons info.  Fixed.  :+1:

**Pull request merged**